core/src/main/java/org/mule/runtime/core/internal/util/message/MessageUtils.java

```diff
@@ -82,8 +85,10 @@ public static Message toMessage(Result result, CursorProviderFactory cursorProvi
    * @param event Used for the case where a {@link CursorProvider} is created, register the one in it.
    * @return a {@link Message}
    */
-  public static Message toMessage(Result result, CursorProviderFactory cursorProviderFactory, CoreEvent event) {
-    return toMessage(result, cursorProviderFactory, ((BaseEventContext) event.getContext()).getRootContext());
+  public static Message toMessage(Result result, CursorProviderFactory cursorProviderFactory, CoreEvent event,
+                                  ComponentLocation originatingLocation) {
+    return toMessage(result, cursorProviderFactory, ((BaseEventContext) event.getContext()).getRootContext(),
```

Comment:
can you make the implementation null safe and pass no value instead of a wrong value for this case?